### PR TITLE
Allow stemmers in languages other than English

### LINKED ISSUE #334: Stemming another languages
Hi! I found that there is already out of box fuzzy search, now there is the only need in morphological search to allow tantivy usage as a regular search engine for general purposes.
I found that you are using rust-stemmers library which already supports a lot of languages, but the only one allowed and used is English. What's wrong with another languages and is it possible to support another now?

### PR DESCRIPTION
**Pull request by pentlander**
Allow users to create stemmers for languages other than English. Add a
default stemmer for English. Closes #334.

### CODE REVIEW

File: src/tokenizer/mod.rs [hunk 1]
```diff
@@ -214,13 +222,41 @@ pub mod tests {
...
+                .filter(Stemmer::new(StemmerAlgorithm::Spanish)),
+        );
+        let en_tokenizer = tokenizer_manager.get("es_stem").unwrap();
+        let mut tokens: Vec<Token> = vec![];
+        {
+            let mut add_token = |token: &Token| {
+                tokens.push(token.clone());
+            };
+            en_tokenizer
+                .token_stream("Hola, feliz contribuyente!")
```
Comment (fulmicoton) [inline src/tokenizer/mod.rs:250]: Ha ha ha :)

File: src/tokenizer/stemmer.rs [hunk 2]
```diff
@@ -1,9 +1,56 @@
...
+#![cfg_attr(feature = "cargo-clippy", allow(clippy::new_without_default, doc))]
 
 use super::{Token, TokenFilter, TokenStream};
 use rust_stemmers::{self, Algorithm};
 use std::sync::Arc;
 
+/// Available stemmer algorithms for different languanges.
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Copy, Clone)]
+#[allow(missing_docs)]
+pub enum StemmerAlgorithm {
```
Comment (fulmicoton) [inline src/tokenizer/stemmer.rs]: Maybe call that language?

Comment (pentlander) [reply src/tokenizer/stemmer.rs]: Will do

**Comment by pentlander** [comment]
Should I do anything about the drop in code coverage? I'm not sure that adding a test for every language really adds value.

File: src/tokenizer/stemmer.rs [hunk 3]
```diff
@@ -1,9 +1,56 @@
-#![cfg_attr(feature = "cargo-clippy", allow(clippy::new_without_default))]
+#![cfg_attr(feature = "cargo-clippy", allow(clippy::new_without_default, doc))]
```
Comment (fulmicoton) [inline src/tokenizer/stemmer.rs]: Do we need that?

**Review by fulmicoton — COMMENTED** [review]
@pentlander Don't worry about the code coverage.

File: src/tokenizer/stemmer.rs (hunk 3 above)
Comment (pentlander) [reply src/tokenizer/stemmer.rs]: Whoops, I didn't mean to leave that in, I'll remove it.

**Pull request merged**